src/do.c

```diff
@@ -2060,12 +2060,16 @@ zombify_mon(anything *arg, long timeout UNUSED)
     int zmon = zombie_form(&mons[body->corpsenm]);
 
     if (zmon != NON_PM) {
-
         if (has_omid(body))
             free_omid(body);
         if (has_omonst(body))
             free_omonst(body);
 
+        /* Ensure we get a new oeaten proportional to the new zombie form's
+         * total possible nutrition, so that the ratio stays the same and can
+         * be used by eaten_stat to determine the zombie's HP. */
+        body->oeaten = (unsigned) (mons[zmon].cnutrit
+                                    * (body->oeaten / obj_nutrition(body)));
```

Comment:
I don't remember discussing this. Is it a vanilla bug?

Reply:
Can corpses be revived as zombies in vanilla?  If so it may be a vanilla bug... I just woke up and can't remember whether that's possible or not.

Reply:
I broke this commit out and moved it over to NetHack/NetHack#497